pinot-plugins/pinot-minion-tasks/pinot-minion-builtin-tasks/src/main/java/org/apache/pinot/plugin/minion/tasks/mergerollup/MergeRollupTaskGenerator.java

```diff
@@ -305,9 +326,12 @@ public List<PinotTaskConfig> generateTasks(List<TableConfig> tableConfigs) {
             long endTimeMs = preSelectedSegment.getEndTimeMs();
             if (endTimeMs >= bucketStartMs) {
               // For segments overlapping with current bucket, add to the result list
-              if (!isMergedSegment(preSelectedSegment, mergeLevel)) {
+              if (!isMergedSegment(preSelectedSegment, mergeLevel, sortedMergeLevels)) {
```

Comment:
Should we compute sorted mergelevels closer to this statement as there is lots of code before? and no earlier use?

Reply:
Moved the computation right before emitting the metrics since it's also an input for the metric.